sugondat/shim/src/cmd/query/mod.rs

```diff
@@ -16,6 +16,31 @@ pub async fn run(params: Params) -> anyhow::Result<()> {
     Ok(())
 }
 
+pub async fn get_block_at(
+    client: &sugondat_rpc::Client,
+    block: BlockParams,
+) -> anyhow::Result<sugondat_rpc::Block> {
+    let maybe_hash = match block.block_ref {
+        BlockRef::Best => None,
+        BlockRef::Hash(h) => Some(h),
+        BlockRef::Number(n) => Some(match block.wait {
+            true => client.wait_finalized_height(n).await,
+            false => {
+                client
+                    .block_hash(n)
+                    .await?
+                    .ok_or_else(|| anyhow::anyhow!("No block with number {}", n))?
+                    .0
+            }
+        }),
+    };
+
+    match block.wait {
+        true => client.wait_block_at(maybe_hash).await,
+        false => client.get_block_at(maybe_hash).await,
+    }
```

Comment:
Is it me, or those matches on booleans look weird?

Reply:
they look weird I agree, but on the other side the if else block is one line longer (incredible) and feel more boring

```rust
    if block.wait {
        client.wait_block_at(maybe_hash).await
    } else {
        client.get_block_at(maybe_hash).await
    }

```

I don't know, I just line more match on boolean than standard if else 😅

Reply:
I used to use the `match boolean` syntax but these days I favor the `if boolean {} else {}` syntax as it's more familiar. I think `match boolean` is OK as long as both branches are 1 line. Then it ends up being more terse & very readable.